app/models/concerns/has_many_common.rb

```diff
@@ -82,7 +82,9 @@ def belongs_to_name_for(association, options)
 
       # SETTER _name= method
       define_method "#{assoc_name}=" do |name_value|
-        assoc_id = assoc_klass(association).send("find_by_#{assoc_klass(association).attribute_name}", name_value).id
+        assoc_id = assoc_klass(association).send("find_by_#{assoc_klass(association).attribute_name}", name_value).try(:id)
```

Comment:
looks like something went wrong in git rebase :) this should not be part of your pr